File: BaseLibrary/Model/QuarantineEntry.cs
using System;

namespace Xvirus.Model;

public class QuarantineEntry
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string OriginalFileName { get; set; } = string.Empty;
    public string OriginalFilePath { get; set; } = string.Empty;
    public string QuarantinedFileName { get; set; } = string.Empty;
    public string OriginalAttributes { get; set; } = string.Empty;
    public DateTime OriginalCreationTime { get; set; }
    public DateTime OriginalLastWriteTime { get; set; }
    public DateTime OriginalLastAccessTime { get; set; }
}

File: XvirusUI/src/mainview/model/QuarantineEntry.ts
export interface QuarantineEntry {
  id: string;
  originalFileName: string;
  originalFilePath: string;
  quarantinedFileName: string;
  originalAttributes: string;
  originalCreationTime: string; // ISO date string
  originalLastWriteTime: string; // ISO date string
  originalLastAccessTime: string; // ISO date string
}
